deposit/declaration.py

```diff
@@ -0,0 +1,256 @@
+import io
+import logging
+import os
+import sys
+
+from copy import copy
+from datetime import date
+
+from reportlab.lib.pagesizes import A4
+from reportlab.lib.styles import ParagraphStyle
+from reportlab.lib.units import mm
+
+from reportlab.platypus import Paragraph
+
+from reportlab.pdfbase import pdfmetrics
+from reportlab.pdfbase.ttfonts import TTFont
+
+from reportlab.pdfgen import canvas
+
+from dissemin.settings import BASE_DIR
+
+logger = logging.getLogger('dissemin.' + __name__)
+
+# Get the current module name to load functions by name
+current_module = sys.modules[__name__]
+
+# The dict contains the available functions and an admin-friendly name
+REGISTERED_DECLARATION_FUNCTIONS = {
+    'declaration_ulb_darmstadt' : 'ULB Darmstadt',
+}
+
+
+def get_declaration_pdf(deposit_record, user):
+    """
+    This function creates for a given deposit the letter of declaration. If the does not succeed, it raises an exception.
+    :param deposit: DepositRecord containing information for declaration
+    :returns: BytesIO containing pdf or raises Exception
+    """
+    func = get_declaration_function(deposit_record.repository.letter_declaration)
+    return func(deposit_record, user)
+
+
+def get_declaration_function(func_name):
+    """
+    This function gets the function to generate a letter of declaration. If no function can be found, we raise AttributeError.
+    :param func_name: Name of the function that is requested
+    :returns: function or raises AttributeError
+    """
+    return getattr(current_module, func_name)
```

Comment:
I personally don't like this much. Why not listing the declaration functions directly within the `REGISTERED_DECLARATION_FUNCTIONS` dict?

Reply:
I was thinking the wrong way and was (once again) not aware of how awesome python can be.

Thank you!